jvcprojector/projector.py

```diff
@@ -101,50 +104,147 @@ async def disconnect(self) -> None:
 
     async def get_info(self) -> dict[str, str]:
         """Get device info."""
-        assert self._device
-        model = JvcCommand(command.MODEL, True)
-        mac = JvcCommand(command.MAC, True)
-        await self._send([model, mac])
+        if not self._device:
+            raise JvcProjectorError("Must call connect before getting info")
+
+        model = JvcCommand(const.CMD_MODEL, True)
+        # ensure PJ is fully on first
+        if await self.is_on():
+            mac = JvcCommand(const.CMD_LAN_SETUP_MAC_ADDRESS, True)
+            version = JvcCommand(const.CMD_VERSION, True)
+            await self._send([model, mac, version])
+        else:
+            await self._send([model])
 
         if mac.response is None:
-            raise JvcProjectorError("Mac address not available")
+            mac.response = "(unknown)"
 
         if model.response is None:
             model.response = "(unknown)"
 
+        # store to reference in state
         self._model = model.response
         self._mac = mac.response
+        self._version = version.response
```

Comment:
pushed a change lmk if thats what you had in mind

```if version.response is None:
            version.response = "(unknown)"
```
otherwise it will default to none

Reply:
The issue is `version` is not defined in line 128 if the projector is off because in line 112 it's skipped. I think its ok to just always get the version from the projector.

Reply:
Also the string `(unknown)` will fail the `process_version()` parse further down.